server/src/main/java/org/apache/iotdb/db/mpp/plan/analyze/PartitionCache.java

```diff
@@ -0,0 +1,757 @@
+   *
+   * @param storageGroupNames the storage groups that need to update
+   */
+  public void updateStorageCache(Set<String> storageGroupNames) {
+    storageGroupCacheLock.writeLock().lock();
+    try {
+      storageGroupCache.addAll(storageGroupNames);
+    } finally {
+      storageGroupCacheLock.writeLock().unlock();
+    }
+  }
+
+  /**
+   * invalid storage group cache
+   *
+   * @param storageGroupNames the storage groups that need to invalid
+   */
+  public void invalidStorageGroupCache(List<String> storageGroupNames) {
+    storageGroupCacheLock.writeLock().lock();
+    try {
+      for (String storageGroupName : storageGroupNames) {
+        storageGroupCache.remove(storageGroupName);
+      }
+    } finally {
+      storageGroupCacheLock.writeLock().unlock();
+    }
+  }
+
+  /** invalid all storage group cache */
+  public void invalidAllStorageGroupCache() {
+    storageGroupCacheLock.writeLock().lock();
+    try {
+      storageGroupCache.clear();
+    } finally {
+      storageGroupCacheLock.writeLock().unlock();
+    }
+  }
+
+  // endregion
+
+  // region replicaSet cache
+  /** get regionReplicaSet from consensusGroupId */
+  public TRegionReplicaSet getRegionReplicaSet(TConsensusGroupId consensusGroupId) {
+    TRegionReplicaSet regionReplicaSet;
+    if (!groupIdToReplicaSetMap.containsKey(consensusGroupId)) {
+      try (ConfigNodeClient client =
+          configNodeClientManager.borrowClient(ConfigNodeInfo.partitionRegionId)) {
+        TRegionRouteMapResp resp = client.getLatestRegionRouteMap();
+        if (TSStatusCode.SUCCESS_STATUS.getStatusCode() == resp.getStatus().getCode()) {
+          updateGroupIdToReplicaSetMap(resp.getTimestamp(), resp.getRegionRouteMap());
+        }
+        if (!groupIdToReplicaSetMap.containsKey(consensusGroupId)) {
+          throw new RuntimeException(
+              "Failed to get replicaSet of consensus group[id= " + consensusGroupId + "]");
+        }
+      } catch (IOException | TException e) {
+        throw new StatementAnalyzeException(
+            "An error occurred when executing getRegionReplicaSet():" + e.getMessage());
+      }
+    }
+    regionReplicaSet = groupIdToReplicaSetMap.get(consensusGroupId);
+    if (null == regionReplicaSet) {
+      throw new RuntimeException(
+          "ReplicaSet of consensus group[id= " + consensusGroupId + "] is null.");
+    }
+    return regionReplicaSet;
+  }
+
+  /** update regionReplicaSetMap according to timestamp */
+  public boolean updateGroupIdToReplicaSetMap(
+      long timestamp, Map<TConsensusGroupId, TRegionReplicaSet> map) {
+    boolean result = (timestamp == latestUpdateTime.accumulateAndGet(timestamp, Math::max));
+    // if timestamp is greater than latestUpdateTime, then update
+    if (result) {
+      groupIdToReplicaSetMap.clear();
+      groupIdToReplicaSetMap.putAll(map);
+    }
+    return result;
+  }
+
+  /** invalid all replica Cache */
+  public void invalidReplicaCache() {
```

Comment:
Suggest to rename to `invalidReplicaSetCache()`

Reply:
Get it